rocketpy/simulation/flight_data_importer.py

```diff
@@ -0,0 +1,399 @@
+    """
+
+    def __init__(
+        self,
+        name,
+        paths,
+        columns_map=None,
+        units=None,
+        interpolation="linear",
+        extrapolation="zero",
+        separator=",",
+        encoding="utf-8",
+    ):
+        """Initializes the FlightDataImporter class.
+
+        Parameters
+        ----------
+        name : str
+            The name of the FlightDataImporter object.
+        paths : list
+            A list of paths or strings representing the file paths or directories
+            containing the flight data files. Only .csv and .txt files are supported.
+        columns_map : dict, optional
+            A dictionary mapping column names to desired column names.
+            Defaults to None, which will keep the original column names.
+        units : dict, optional
+            A dictionary mapping column names to desired units.
+            Defaults to None, which will consider that all the data is in SI.
+        interpolation : str, optional
+            The interpolation method to use for missing data.
+            Defaults to "linear", see rocketpy.mathutils.Function for more
+            information.
+        extrapolation : str, optional
+            The extrapolation method to use for data outside the range.
+            Defaults to "zero", see rocketpy.mathutils.Function for more
+            information.
+        separator : str, optional
+            The separator used in the data file. Defaults to ",".
+        encoding : str, optional
+            The encoding of the data file. Defaults to "utf-8".
+
+        Notes
+        -----
+        Try to avoid using forbidden characters in the column names, such as
+        parenthesis, brackets, etc. These characters may cause errors when
+        accessing the attributes of the Flight object.
+        """
+        self.name = name
+        self.paths = paths
+
+        # Initialize debuggers
+        self._columns_map = {}
+        self._units = {}
+        self._columns = {}
+        self._time_cols = {}
+        self._original_columns = {}
+        self._data = {}
+        self._separators = {}
+        self._encodings = {}
+        self._files = None
+
+        # So now we are going to loop through the files and read them
+        self.read_data(
+            paths,
+            columns_map,
+            units,
+            interpolation,
+            extrapolation,
+            separator,
+            encoding,
+        )
+
+    def __repr__(self):
+        """Representation method for the FlightDataImporter class.
+
+        Returns
+        -------
+        str
+            A string representation of the FlightDataImporter class.
+        """
+        return f"FlightDataImporter(name='{self.name}', paths='{self.paths}')"
+
+    def __reveal_files(self, path):
+        """Get a list of all the .csv or .txt files in the given path, or simply
+        return the path of the file if it is a file.
```

Comment:
Do we follow any conventions when naming methods meant to be private? Sometimes we do it with one underscore, sometimes we do it with two.

From what I've researched it is more common to use only one underscore for the use case of making a method "private" or "protected"

Reply:
Writing "Class._method" and "Class.__method" are not the same thing.

Private methods in python actually doesn't exist.
Using double underscore will make python "to mangle" the attribute, which is basically adding a "__ClassName__" at the beginning of the attribute's name, which happens to be a uncomfortable way of "privatizing" attributes and methods within a class.

Using a single underscore is the informal way python developers found to define "private methods" without mangling its name.

I left these methods with double underscores because I don't think users (or developers) should invoke these methods out of this class. I mean, I wanted them private, not "weakly private".